src/lua/crypto_bbs.lua

```diff
@@ -644,16 +644,19 @@ local function core_proof_gen(ciphersuite, pk, signature, generators, header, ph
     local signature_result = octets_to_signature(signature)
     local AA, e = table.unpack(signature_result)
     local undisclosed_indexes = {}
+    local disclosed_messages = {}
+    local undisclosed_messages = {}
+    for i, v in ipairs(disclosed_indexes) do
+        if i > 1 and disclosed_indexes[i] == disclosed_indexes[i - 1] then
```

Comment:
I am not sure that checking only adjacent values suffices here. There can be scattered duplicates.

Reply:
Arrays are always sorted in proof_gen (line 684) before being passed as input to core_proof_gen

Reply:
can you add this sentence as comment in that line of code?